third_party/blink/renderer/modules/brave/navigator_browsertest.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "base/path_service.h"
+#include "brave/browser/brave_content_browser_client.h"
+#include "brave/common/brave_paths.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/common/chrome_content_client.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "content/public/test/browser_test_utils.h"
+
+namespace {
+
+const char kDetectBraveTest[] = "/detect_brave.html";
+
+class NavigatorGetBraveDetectedTest : public InProcessBrowserTest {
+ public:
+  void SetUpOnMainThread() override {
+    InProcessBrowserTest::SetUpOnMainThread();
+
+    content::SetupCrossSiteRedirector(embedded_test_server());
+
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    embedded_test_server()->ServeFilesFromDirectory(test_data_dir);
+
+    ASSERT_TRUE(embedded_test_server()->Start());
+  }
+};
+
+IN_PROC_BROWSER_TEST_F(NavigatorGetBraveDetectedTest, IsDetected) {
+  GURL url = embedded_test_server()->GetURL(kDetectBraveTest);
+  ui_test_utils::NavigateToURL(browser(), url);
+  content::WebContents* contents =
+      browser()->tab_strip_model()->GetActiveWebContents();
+  EXPECT_EQ(true, EvalJs(
```

Comment:
`EXPECT_TRUE(EvalJs(contents, "getBraveDetected()"));`

Reply:
I tried to do this when you had shared feedback before; it's causing a problem though:
```
../../third_party/googletest/src/googletest/include/gtest/gtest.h:299:9: error: no viable conversion from 'const content::EvalJsResult' to 'bool'
```

Reply:
From `browser_test_utils.h`:

```
//      For boolean results, note that EXPECT_TRUE(..) and EXPECT_FALSE()
//      won't compile; use EXPECT_EQ(true, ...) instead. This is intentional,
//      since EXPECT_TRUE() could be read ambiguously as either "expect
//      successful execution", "expect truthy value of any type", or "expect
//      boolean value 'true'".
```

Reply:
ah- thanks, @zenparsing! 😄